src/viking.js

```diff
@@ -1,11 +1,94 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`;
+    } else return `${this.name} has died in act of combat`;
+  }
+
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`;
+    } else return `A Saxon has died in combat`;
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+
+  addViking(aVikingObject) {
+    this.vikingArmy.push(aVikingObject);
+  }
+
+  addSaxon(aSaxonObject) {
+    this.saxonArmy.push(aSaxonObject);
+  }
+
+  vikingAttack() {
+    return this.attack(this.vikingArmy, this.saxonArmy);
+  }
+
+  saxonAttack() {
+    return this.attack(this.saxonArmy, this.vikingArmy);
+  }
+
+  removeDeadSoldier(whatArmy) {
+    whatArmy.forEach((soldier, index) => {
+      if (soldier.health < 1) whatArmy.splice(index, 1);
+    });
+  }
+
+  attack(offense, defense) {
+    const offSoldier = this.random(offense);
+    const defSoldier = this.random(defense);
+    const fight = defSoldier.receiveDamage(offSoldier.strength);
+    this.removeDeadSoldier(defense);
+    return fight;
+  }
+
+  showStatus() {
+    if (this.saxonArmy.length === 0) {
```

Comment:
you can drop the elses down there. 
it's called "early" return pattern and most people think it's easier to read than nested if elses. you might even be able to inline some of the returns and not give them their own {} code blocks :)